wrappers/csharp/Intel.RealSense/Intel.RealSense.2015.csproj

```diff
@@ -75,20 +75,55 @@
     <BuildPlatform>$(Platform)</BuildPlatform>
-    <BuildPlatform Condition="'$(BuildPlatform)' == 'AnyCPU'">x86</BuildPlatform>
+    <BuildPlatform Condition="'$(Platform)'=='AnyCPU'">x86</BuildPlatform>
+    <BuildPlatform Condition="'$(Platform)'=='AnyCPU' and '$(PROCESSOR_ARCHITEW6432)'=='AMD64'">x64</BuildPlatform>
     <InstallPath>C:\Program Files (x86)\Intel RealSense SDK 2.0\bin</InstallPath>
+    <BuildPathx86></BuildPathx86>
+    <BuildPathx64></BuildPathx64>
     <BuildPath Condition="'$(BuildPath)'==''">..\..\..\build\$(Configuration)\realsense2.dll</BuildPath>
   </PropertyGroup>
-  <Target Name="BeforeBuild">
-    <CallTarget Targets="BuildLib" ContinueOnError="true" />
-    <CallTarget Targets="CopyLocalDependencies" />
+
+  <Target Name="BeforeBuild" DependsOnTargets="IncludeLib">
+  </Target>
+
+  <Target Name="IncludeLib" DependsOnTargets="EnsureBitCorrectness">
+    <PropertyGroup>
+      <LibFile Condition="'$(BuildPlatform)'=='x86'">$(LibPathx86)</LibFile>
+      <LibFile Condition="'$(BuildPlatform)'=='x64'">$(LibPathx64)</LibFile>
+    </PropertyGroup>
+    <Copy Condition="'$(LibFile)'!=''" SourceFiles="$(LibFile)" DestinationFolder="$(OutputPath)" />
+    <Message Condition="'$(LibFile)'!=''" Text="realsense2.dll in output folder from $(LibFile)" Importance="high"/>
   </Target>
-  <Target Name="CopyLocalDependencies" BeforeTargets="Build" Condition="'$(IsCrossTargetingBuild)' != 'true'">
-    <Copy Condition="Exists('$(BuildPath)')" SourceFiles="$(BuildPath)" DestinationFolder="$(OutputPath)" />
-    <Copy Condition="!Exists('$(BuildPath)') and Exists('$(InstallPath)')" SourceFiles="$(InstallPath)\$(BuildPlatform)\realsense2.dll" DestinationFolder="$(OutputPath)" />
+
+  <Target Name="EnsureBitCorrectness">
+    <CallTarget Targets="BuildLib" ContinueOnError="true" />
+    <Exec Condition="'$(BuildPathx86)'=='' and '$(BuildPathx64)'=='' and Exists($(BuildPath))" ConsoleToMSBuild="true" StandardOutputImportance="low" Command="call &quot;$(DevEnvDir)..\Tools\VsDevCmd.bat&quot; &amp;&amp; dumpbin.exe /headers $([System.IO.Path]::GetFullPath($(BuildPath)))">
+      <Output TaskParameter="ConsoleOutput" PropertyName="OutputOfExec" />
+    </Exec>
+    <PropertyGroup>
+      <LibArch Condition="$(OutputOfExec.Contains('machine (x86)'))">x86</LibArch>
+      <LibArch Condition="$(OutputOfExec.Contains('machine (x64)'))">x64</LibArch>
+      <BuildPathx86 Condition="'$(BuildPathx86)'=='' and '$(LibArch)'=='x86'">$(BuildPath)</BuildPathx86>
+      <BuildPathx64 Condition="'$(BuildPathx64)'=='' and '$(LibArch)'=='x64'">$(BuildPath)</BuildPathx64>
+      <LibPathx86 Condition="Exists('$(InstallPath)')">$(InstallPath)\x86\realsense2.dll</LibPathx86>
+      <LibPathx64 Condition="Exists('$(InstallPath)')">$(InstallPath)\x64\realsense2.dll</LibPathx64>
+      <LibPathx86 Condition="Exists('$(BuildPathx86)')">$(BuildPathx86)</LibPathx86>
+      <LibPathx64 Condition="Exists('$(BuildPathx64)')">$(BuildPathx64)</LibPathx64>
```

Comment:
Don't these lines (111-114) gives priority to installed version? (I ask again because on my machine it took the installed one even though I had a local build)

Reply:
They are intended to set the installed paths as backups and then overwrite them with the local build paths if they exist. Are your local build paths set in the BuildPathx[86/64] properties (lines 84/85 in 2015)? Do they include the file name `realsense2.dll`? Are there any messages about compatibility issues?

Reply:
I'll try again when I'm next to a machine and let you know